python/pyspark/sql/connect/plan.py

```diff
@@ -1320,20 +1396,7 @@ def plan(self, session: "SparkConnectClient") -> proto.Relation:
             for k, arg in self._named_args.items():
                 plan.sql.named_arguments[k].CopyFrom(arg.to_plan(session))
 
-        if self._views is not None and len(self._views) > 0:
-            # build new plan like
-            # with_relations [id 10]
-            #     root: sql  [id 9]
-            #     reference:
-            #          view#1: [id 8]
-            #          view#2: [id 5]
-            sql_plan = plan
-            plan = proto.Relation()
-            plan.common.plan_id = self._plan_id_with_rel
-            plan.with_relations.root.CopyFrom(sql_plan)
-            plan.with_relations.references.extend([v.plan(session) for v in self._views])
-
-        return plan
+        return self._with_relations(plan, [v.plan(session) for v in self._references])
```

Comment:
Just to make sure this PR won't change the protobuf of `spark.sql`, right?

Reply:
No, this won't change it.